op-batcher/batcher/espresso.go

```diff
@@ -39,6 +44,47 @@ const (
 	finalityCheckInterval = 100 * time.Millisecond
 )
 
+func (l *BatchSubmitter) registerBatcher(ctx context.Context) error {
+	if l.Attestation == nil {
+		l.Log.Warn("Attestation is nil, skipping registration")
+		return nil
+	}
+
+	batchInbox, err := bindings.NewBatchInbox(l.RollupConfig.BatchInboxAddress, l.L1Client)
+	if err != nil {
+		return fmt.Errorf("failed to create batch inbox contract bindings: %w", err)
+	}
+
+	// Decode the attestation off-chain to conserve gas
+	decoded, err := batchInbox.DecodeAttestationTbs(&bind.CallOpts{}, l.Attestation)
```

Comment:
Will this attestation contain the public key corresponding to the private key in the `public key` field?